openlmis-stock/src/main/java/org/smartregister/stock/openlmis/presenter/StockDetailsPresenter.java

```diff
@@ -217,4 +221,40 @@ private boolean processStock(String step, JSONObject jsonString, String provider
         return true;
     }
 
+    private boolean processStockWithoutLots(JSONObject jsonString, String provider, String date,
+                                            String facility, String reason, String transactionType, int quantity) throws JSONException {
+
+        JSONArray stepFields = jsonString.getJSONObject(STEP1).getJSONArray(FIELDS);
+
+        String tradeItem = extractTradeItemId(stepFields);
+
+        Date encounterDate;
+        try {
+            encounterDate = simpleDateFormat.parse(date);
+        } catch (ParseException e) {
+            Log.e(TAG, "error passing stock issue/received date", e);
+            encounterDate = new Date();
+        }
+
+        Stock stock = new Stock(null, transactionType,
+                provider, quantity,
+                encounterDate.getTime(), facility, TYPE_Unsynced,
+                System.currentTimeMillis(), tradeItem);
+        stock.setReason(reason);
+        totalStockAdjustment += stock.getValue();
+        stockDetailsInteractor.addStock(stock);
+
+        return true;
+    }
+
+    private String extractTradeItemId(JSONArray stepFields) throws JSONException {
```

Comment:
Remove this method it was refactored to more generic extractValue and passing second param as key to be extracted